### PR TITLE
Verify image was produced by a trusted task

### PR DESCRIPTION
**Pull request by lcarva**
Ref: EC-253

### CODE REVIEW

File: policy/release/slsa_build_scripted_build.rego [hunk 1]
```diff
@@ -120,3 +152,15 @@ subject_digest(subject) := digest if {
...
+trusted_build_task_error(build_task_bundles) := error if {
+	count(build_task_bundles) == 0
+	error := "No Pipeline Tasks built the image"
+} else := error if {
+	untrusted := [task |
+		some task in bundles.unacceptable_task_bundle(lib.tasks_from_pipelinerun)
+		task.ref.bundle in build_task_bundles
+	]
+	count(untrusted) > 0
+	error := "Build Task is not trusted"
```
Comment (zregvart) [inline policy/release/slsa_build_scripted_build.rego:165]: Should the error say which task(s) were this?
```suggestion
	error := sprintf("Build Task(s) %v are not trusted", [untrusted])
```

File: policy/release/slsa_build_scripted_build.rego [hunk 2]
```diff
@@ -112,6 +113,37 @@ deny contains result if {
...
+#
+deny contains result if {
+	expected_digest := input.image.ref
+
+	# Find all the Tekton Bundle references from the Tasks that claim to have built the image being
+	# validated.
+	bundles := {bundle |
+		some attestation in lib.pipelinerun_attestations
+		some build_task in tkn.build_tasks(attestation)
+		digest := tkn.task_result(build_task, "IMAGE_DIGEST")
```
Comment (zregvart) [inline policy/release/slsa_build_scripted_build.rego:138]: Do we want to support [other types](https://tekton.dev/docs/chains/slsa-provenance/#output-artifacts) of results here as well?

File: policy/release/slsa_build_scripted_build.rego (hunk 1 above)
Comment (lcarva) [reply policy/release/slsa_build_scripted_build.rego]: Yes, much better!

File: policy/release/slsa_build_scripted_build.rego (hunk 2 above)
Comment (lcarva) [reply policy/release/slsa_build_scripted_build.rego:139]: We should. But adding support here won't really help as the underlying utility functions, e.g. `tkn.build_tasks`, don't support them. Filed https://github.com/enterprise-contract/ec-policies/issues/840 so we can address this properly.

**Review by zregvart — APPROVED** [review]

**Review by simonbaird — APPROVED** [review]
Lgtm.

**Pull request merged**